src/read.c

```diff
@@ -789,14 +790,19 @@ static avifResult avifMetaFindOrCreateItem(avifMeta * meta, uint32_t itemID, avi
     assert(itemID != 0);
 
     for (uint32_t i = 0; i < meta->items.count; ++i) {
-        if (meta->items.item[i].id == itemID) {
-            *item = &meta->items.item[i];
+        if (meta->items.item[i]->id == itemID) {
+            *item = meta->items.item[i];
             return AVIF_RESULT_OK;
         }
     }
 
-    *item = (avifDecoderItem *)avifArrayPush(&meta->items);
-    AVIF_CHECKERR(*item != NULL, AVIF_RESULT_OUT_OF_MEMORY);
+    avifDecoderItem ** item_ptr = (avifDecoderItem **)avifArrayPush(&meta->items);
+    AVIF_CHECKERR(item_ptr != NULL, AVIF_RESULT_OUT_OF_MEMORY);
+    *item_ptr = (avifDecoderItem *)avifAlloc(sizeof(avifDecoderItem));
+    AVIF_CHECKERR(*item_ptr != NULL, AVIF_RESULT_OUT_OF_MEMORY);
+    memset(*item_ptr, 0, sizeof(avifDecoderItem));
+
+    *item = *item_ptr;
```

Comment:
Nit/optional: We may want to take the opportunity to set `*item` back to `NULL` before the two error returns, so that we never leave a dangling pointer in `*item`, even when this function returns an error.

Reply:
it's not really necessary because the caller honors the failure return code. but done to be safe.